Create footer links 'Main,' 'Information For,' 'Topics,' & 'Resources'

smayden opened the pull request:
This ticket creates and styles four of the menus/blocks in the footer and subfooter. 

## Description
This PR adds footer menus/blocks for:
-Main (footer region)
-Information For (subfooter)
-Topics (subfooter)
-Resources (subfooter)

It also has twig templates for styling links in the footer/subfooter regions and for each block (for styling the heading).

## Related Issues
#579 
#580 
#581 
#582 

## How Has This Been Tested?
Tested locally and in Probo

To test:

Add links to each menu: /admin/structure/menu
Link should appear in the sub footer region

![image](https://user-images.githubusercontent.com/10873398/117490927-57b38b80-af24-11eb-8098-ad7be48813bd.png)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

joeparsons commented on config/install/system.menu.az-footer-information-for.yml:
```diff
@@ -0,0 +1,7 @@
+dependencies: {  }
+id: az-footer-information-for
+label: 'Footer - Information For'
+description: 'Generally contains enterprise level navigation links to audience-based sections of Arizona.edu (e.g. current students, faculty).'
```
Nitpick: institution vs enterprise?
```suggestion
description: 'Generally contains institution level navigation links to audience-based sections of Arizona.edu (e.g. current students, faculty).'
```

smayden replied:
Changed the language here

smayden commented:
Remaining styling:
-Topics menu should wrap to two columns. Looks like we did this with custom css in QS1:
https://bitbucket.org/ua_drupal/ua_quickstart/src/7.x-1.x/modules/custom/uaqs_navigation/uaqs_navigation_global_footer/css/uaqs_navigation_global_footer.css
-Do we want to keep icons for the Resources? How should that be implemented?

Looking ahead:
-Have questions on best implementation for social media (separate module in QS1)

smayden commented:
Added the split-2-col class to topics menu That Dana had added to Bootstrap ([#276](https://github.com/az-digital/arizona-bootstrap/pull/276)

Can test by adding the css in the inspector:

.split-2-col {
    display: block;
    column-count: 2;
}

![image](https://user-images.githubusercontent.com/10873398/117744676-01617980-b1be-11eb-840b-2d55439352d5.png)

danahertzberg commented on themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig:
```diff
@@ -0,0 +1,62 @@
+  {% import _self as menus %}
+  {% if items %}
+    {% if menu_level == 0 %}
+      <ul{{ attributes.addClass('nav').addClass('nav-utility').addClass('d-flex').addClass('justify-content-end') }}>
```
Might want to consider adding these classes to have this menu stack on mobile.

```suggestion
      <ul{{ attributes.addClass('nav').addClass('nav-utility').addClass('d-flex').addClass('justify-content-end').addClass('flex-column').addClass('flex-md-row') }}>
```

danahertzberg commented on themes/custom/az_barrio/templates/navigation/menu--footer_main.html.twig:
```diff
@@ -0,0 +1,62 @@
+      <li{{ item.attributes.addClass(classes) }}>
+        {%
+          set link_classes = [
+            'nav-link',
```
Add `d-flex` class to have hover underline only the word

```suggestion
            'nav-link d-flex',
```

danahertzberg commented on themes/custom/az_barrio/templates/navigation/menu--sub_footer_topics.html.twig:
```diff
@@ -0,0 +1,62 @@
+  {% import _self as menus %}
+  {% if items %}
+    {% if menu_level == 0 %}
+      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-utility').addClass('split-2-col') }}>
```
I will need to update `split-2-col` in Bootstrap to apply only on medium screens and up

danahertzberg commented on themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-info.html.twig:
```diff
@@ -0,0 +1,57 @@
+  ]
+%}
+{% set heading_id = attributes.id ~ '-menu'|clean_id %}
+<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-4 col-md-3 col-lg-3')|without('role', 'aria-labelledby') }}>
```
Shouldn't need the col-lg-3 class
```suggestion
<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-4 col-md-3')|without('role', 'aria-labelledby') }}>
```

danahertzberg commented on themes/custom/az_barrio/templates/blocks/block--az-barrio-footer-menu-topics.html.twig:
```diff
@@ -0,0 +1,57 @@
+  ]
+%}
+{% set heading_id = attributes.id ~ '-menu'|clean_id %}
+<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-8 col-md-5 col-lg-5')|without('role', 'aria-labelledby') }}>
```
Shouldn't need the col-lg-5 class
```suggestion
<nav role="navigation" aria-labelledby="{{ heading_id }}"{{ attributes.addClass('col-12 col-sm-8 col-md-5')|without('role', 'aria-labelledby') }}>
```

smayden commented:
I made all of Dana's suggested edits and this PR is ready for review.

smayden commented:
This ticket is ready for review. FYI Implemented all of Dana's suggestions during the workshop time (thanks @danahertzberg). 

The topics menu has the split-md-2-col class that Dana plans to introduce in Bootstrap. This should make that menu go to two columns are medium and larger displays.
